Userland/Libraries/LibWeb/Bindings/WindowObject.cpp

```diff
@@ -437,4 +449,167 @@ JS_DEFINE_NATIVE_FUNCTION(WindowObject::inner_height_getter)
+    String behavior_string = "auto";
+
+    if (vm.argument_count() == 1) {
+        auto* options = vm.argument(0).to_object(global_object);
+        if (vm.exception())
+            return {};
+
+        auto behavior_string_value = options->get("behavior");
+        if (vm.exception())
+            return {};
+        behavior_string = behavior_string_value.is_undefined() ? "auto" : behavior_string_value.to_string(global_object);
+        if (vm.exception())
+            return {};
+        if (behavior_string != "smooth" && behavior_string != "auto") {
+            vm.throw_exception<JS::TypeError>(global_object, "Behavior is not one of 'smooth' or 'auto'");
+            return {};
+        }
+
+        auto left = options->get("left");
+        if (vm.exception())
+            return {};
+        if (!left.is_undefined())
+            x_value = left;
+
+        auto top = options->get("top");
+        if (vm.exception())
+            return {};
+        if (!top.is_undefined())
+            y_value = top;
+
+    } else if (vm.argument_count() >= 2) {
+        x_value = vm.argument(0);
+        y_value = vm.argument(1);
+    }
+
+    ScrollBehavior behavior = (behavior_string == "smooth") ? ScrollBehavior::Smooth : ScrollBehavior::Auto;
+
+    double x = x_value.to_double(global_object);
+    if (vm.exception())
+        return {};
+    x = JS::Value(x).is_finite_number() ? x : 0.0;
+
+    double y = y_value.to_double(global_object);
+    if (vm.exception())
+        return {};
+    y = JS::Value(y).is_finite_number() ? y : 0.0;
+
+    // FIXME: Are we calculating the viewport in the way this function expects?
+    // FIXME: Handle overflow-directions other than top-left to bottom-right
+
+    perform_a_scroll(page, x, y, behavior);
+    return JS::js_undefined();
+}
+
+// https://www.w3.org/TR/cssom-view/#dom-window-scrollby
+JS_DEFINE_NATIVE_FUNCTION(WindowObject::scroll_by)
+{
+    auto* impl = impl_from(vm, global_object);
+    if (!impl)
+        return {};
+    if (!impl->page())
+        return {};
+    auto& page = *impl->page();
+
+    JS::Object* options = nullptr;
+
+    if (vm.argument_count() == 0) {
+        options = JS::Object::create(global_object, nullptr);
+    } else if (vm.argument_count() == 1) {
+        options = vm.argument(0).to_object(global_object);
+        if (vm.exception())
+            return {};
+    } else if (vm.argument_count() >= 2) {
+        options = JS::Object::create(global_object, nullptr);
+        options->set("left", vm.argument(0), ShouldThrowExceptions::No);
+        options->set("top", vm.argument(1), ShouldThrowExceptions::No);
+        options->set("behavior", JS::js_string(vm, "auto"), ShouldThrowExceptions::No);
+    }
+
+    auto behavior_string_value = options->get("behavior");
+    if (vm.exception())
+        return {};
+    auto behavior_string = behavior_string_value.is_undefined() ? "auto" : behavior_string_value.to_string(global_object);
```

Comment:
Oh here it is slightly different but you could still do
```suggestion
    String behavior_string = "auto";
    if (!behavior_string_value.is_undefined()) {
        behavior_string = behavior_string_value.to_string(global_object);
        if (vm.exception())
            return {};
    }
```

But I think it's fine either way

Reply:
Thanks. I've kept it this way for scrollBy, and changed it for scroll as you suggested.